de.metas.contracts/src/main/java/de/metas/contracts/pricing/trade_margin/CustomerTradeMarginRepository.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * #%L
+ * de.metas.contracts
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.contracts.pricing.trade_margin;
+
+import com.google.common.collect.ImmutableList;
+import de.metas.bpartner.BPartnerId;
+import de.metas.cache.CCache;
+import de.metas.contracts.model.I_C_Customer_Trade_Margin;
+import de.metas.util.Services;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.ad.dao.impl.CompareQueryFilter;
+import org.springframework.stereotype.Repository;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class CustomerTradeMarginRepository
+{
+
+	private final CCache<CacheKey, ImmutableList<CustomerTradeMarginSettings>> cache = CCache.<CacheKey, ImmutableList<CustomerTradeMarginSettings>> builder()
+			.cacheName("customerTradeMarginBySalesRepAndDate")
+			.cacheMapType(CCache.CacheMapType.LRU)
+			.initialCapacity(100)
+			.additionalTableNameToResetFor(I_C_Customer_Trade_Margin.Table_Name)
```

Comment:
i checked `de.metas.cache.CCache.CCache(String, String, Set<String>, Integer, Integer, CacheMapType, CachingKeysMapper<K>, CacheRemovalListener<K, V>, CacheAdditionListener<K, V>)`..
I think this *will* work, but to reduce FUD, please do
```suggestion
			.tableName(I_C_Customer_Trade_Margin.Table_Name)
```
..and use `additionalTableNameToResetFor` only if there are further names